Rewrote script in Python

tuxudo opened the pull request:
Also reorganized UI views

Commit:
First commit

tuxudo committed:
Fixed condition bug and cleaned up whitespace

tuxudo committed:
Fixed condition bug and cleaned up whitespace

ofirgalcon committed:
Update power_battery_condition_widget.php

Fixes listing

ofirgalcon committed:
Update power_battery_condition_widget.php

Fix listing

bochoven committed:
Merge pull request #6 from ofirgalcon/patch-2

Update power_battery_condition_widget.php

bochoven committed:
Merge pull request #7 from ofirgalcon/patch-3

Update power_battery_condition_widget.php

bochoven commented:
I'm afraid I messed up this PR with my latest code changes. We should check if my edits did not break the data from the python script.

tuxudo committed:
Merged refactor

tuxudo committed:
Merge branch 'master' into python-based

tuxudo commented:
Finally have the conflicts resolved and tested, PR ready to be merged

bochoven commented on power_controller.php:
```diff
@@ -48,15 +48,15 @@ public function get_data($serial_number = '')
-        $out = array();
+        $obj = new View();
+        
         if (! $this->authorized()) {
```
You can get rid of all the $this-authorized() calls. All methods are now protected except get_script() and index()

bochoven committed:
Update power_tab.php

Not tested, but just to show how not to put Mac supplied data inside a html() tag

bochoven reviewed (commented):
Please rework all parts where Mac data is rendered inside .html()

tuxudo committed:
Removed unath calls

tuxudo committed:
Removed .html() rendering

tuxudo commented:
Removed html calls
